src/coreclr/jit/gentree.cpp

```diff
@@ -20337,6 +20371,11 @@ GenTree* Compiler::gtNewSimdCmpOpNode(
                     intrinsic = NI_AVX_CompareGreaterThanOrEqual;
                 }
             }
+            else if (simdSize == 64)
+            {
+                assert(IsBaselineVector512IsaSupportedDebugOnly());
+                intrinsic = NI_AVX512F_CompareGreaterThanOrEqualSpecial;
```

Comment:
Shouldn't this be `NI_AVX512F_CompareGreaterThanOrEqual`

which is then lowered to `NI_AVX512F_CompareGreaterThanOrEqualSpecial` + `NI_AVX512F_MoveMaskToVectorSpecial`?

Similar for the others below.

Reply:
Not exactly

How `Equals()` work for Vector512
NI_Vector512_Equals ->(calls gtNewSimdCmpOpNode(GT_EQ..))->`NI_AVX512F_CompareEqualSpecial `+ `NI_AVX512F_MoveMaskToVectorSpecial`(Both `NI_AVX512F_CompareEqualSpecial `(needs `imm `value based on comparison- added in `genAvxFamilyIntrinsic()`+ `NI_AVX512F_MoveMaskToVectorSpecial` are picked up directly from table)

I'm basically extending this for all comparison operators.

Reply:
Ah, I see. I thought we were doing it in lowering, but we're doing it up in import instead

Thanks, LGTM!